lib/kubeclient/common.rb

```diff
@@ -176,6 +175,23 @@ def self.parse_definition(kind, name)
       )
     end
 
+    def self.resolve_unconventional_method_names(name, kind, singular_name)
+      if name.start_with?(singular_name)
+        plural_suffix = name[kind.downcase.length..-1]             # "es"
+        singular_underscores = ClientMixin.underscore_entity(kind) # "component_status"
```

Comment:
if it makes sense, could you update comments on the right to some realistic example that would get into this function?
I think ComponentStatus / componentstatuses is covered by normal `name.start_with?(prefix.downcase)` flow.